include/wintls/detail/sspi_handshake.hpp

```diff
@@ -74,25 +115,67 @@ class sspi_handshake {
-    if (handshake_type_ == handshake_type::server && server_cert != nullptr) {
-      creds.cCreds = 1;
-      creds.paCred = &server_cert;
-    }
+    bool is_tlsv13 = !!(static_cast<int>(context_.method_) & static_cast<int>(method::tlsv13));
+
+    if (!is_tlsv13) {
+      pCred = &creds;
+
+      creds.dwVersion = SCHANNEL_CRED_VERSION;
+      creds.grbitEnabledProtocols = static_cast<DWORD>(context_.method_);
+      creds.dwFlags = SCH_CRED_MANUAL_CRED_VALIDATION | SCH_CRED_NO_DEFAULT_CREDS;
+      // If revocation checking is enables, specify SCH_CRED_REVOCATION_CHECK_CHAIN_EXCLUDE_ROOT
+      // to cause the TLS certificate status request extension (commonly known as OCSP stapling)
+      // to be sent. This flag matches the CERT_CHAIN_REVOCATION_CHECK_CHAIN_EXCLUDE_ROOT
+      // flag that we pass to the CertGetCertificateChain calls during our manual authentication.
+      if (check_revocation_) {
+        creds.dwFlags |= SCH_CRED_REVOCATION_CHECK_CHAIN_EXCLUDE_ROOT;
+      }
+
+      if (handshake_type_ == handshake_type::server && server_cert != nullptr) {
+        creds.cCreds = 1;
+        creds.paCred = &server_cert;
+      }
+
+      // TODO: rename server_cert field since it is also used for client cert.
+      // Note: if client cert is set, sspi will auto validate server cert with it.
+      // Even though verify_server_certificate_ in context is set to false.
+      if (handshake_type_ == handshake_type::client && server_cert != nullptr) {
+        creds.cCreds = 1;
+        creds.paCred = &server_cert;
+      }
+    } else {
+      pCred = &credentials;
+
+      credentials.dwVersion = SCH_CREDENTIALS_VERSION;
+      credentials.dwFlags = SCH_USE_STRONG_CRYPTO;
+      credentials.cTlsParameters = 1;
+      credentials.pTlsParameters = &tls_parameters;
+      credentials.pTlsParameters->grbitDisabledProtocols = ~static_cast<DWORD>(context_.method_);
+
+      if (check_revocation_) {
+        credentials.dwFlags |= SCH_CRED_REVOCATION_CHECK_CHAIN_EXCLUDE_ROOT;
+      }
+
+      if (handshake_type_ == handshake_type::server && server_cert != nullptr) {
+        credentials.cCreds = 1;
+        credentials.paCred = &server_cert;
+      }
 
-    // TODO: rename server_cert field since it is also used for client cert.
-    // Note: if client cert is set, sspi will auto validate server cert with it.
-    // Even though verify_server_certificate_ in context is set to false.
-    if (handshake_type_ == handshake_type::client && server_cert != nullptr) {
-      creds.cCreds = 1;
-      creds.paCred = &server_cert;
+      // TODO: rename server_cert field since it is also used for client cert.
+      // Note: if client cert is set, sspi will auto validate server cert with it.
+      // Even though verify_server_certificate_ in context is set to false.
+      if (handshake_type_ == handshake_type::client && server_cert != nullptr) {
+        credentials.cCreds = 1;
+        credentials.paCred = &server_cert;
+      }
     }
```

Comment:
Most of the code is duplicated in the different if/else branches here (`is_tlsv3`).

Please find a way to refactor this to avoid that duplication.